src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc

```diff
@@ -0,0 +1,170 @@
+    if (numModes <= 0) {
+        // error
+    }
+    bool found_it = false;
+    for (int modeNum = 1; modeNum <= numModes; ++modeNum) {
+        int NumAlphas; // Number of Alphas for each GetObjectItem call
+        int NumNumbers; // Number of Numbers for each GetObjectItem call
+        int IOStatus;
+        InputProcessor::GetObjectItem(CoilCoolingDXCurveFitOperatingMode::object_name, modeNum, cAlphaArgs, NumAlphas,
+                                      rNumericArgs, NumNumbers, IOStatus);
+        if (!InputProcessor::SameString(name_to_find, cAlphaArgs(1))) {
+            continue;
+        }
+        found_it = true;
+
+        CoilCoolingDXCurveFitOperatingModeInputSpecification input_specs;
+
+        input_specs.name = cAlphaArgs(1);
+        input_specs.gross_rated_total_cooling_capacity = rNumericArgs(1);
+        this->ratedGrossTotalCap = input_specs.gross_rated_total_cooling_capacity;
+        input_specs.rated_evaporator_air_flow_rate = rNumericArgs(2);
+        this->ratedEvapAirFlowRate = input_specs.rated_evaporator_air_flow_rate;
+        input_specs.rated_condenser_air_flow_rate = rNumericArgs(3);
+        input_specs.maximum_cycling_rate = rNumericArgs(4);
+        this->maxCyclingRate = input_specs.maximum_cycling_rate;
+        input_specs.ratio_of_initial_moisture_evaporation_rate_and_steady_state_latent_capacity = rNumericArgs(5);
+        this->evapRateRatio = input_specs.ratio_of_initial_moisture_evaporation_rate_and_steady_state_latent_capacity;
+        input_specs.latent_capacity_time_constant = rNumericArgs(6);
+        this->latentTimeConst = input_specs.latent_capacity_time_constant;
+        input_specs.nominal_time_for_condensate_removal_to_begin = rNumericArgs(7);
+        this->timeForCondensateRemoval = input_specs.nominal_time_for_condensate_removal_to_begin;
+        input_specs.apply_latent_degradation_to_speeds_greater_than_1 = cAlphaArgs(2);
+        input_specs.condenser_type = cAlphaArgs(3);
+        input_specs.nominal_evap_condenser_pump_power = rNumericArgs(8);
+        input_specs.capacity_control = cAlphaArgs(4);
+        input_specs.nominal_speed_number = rNumericArgs(9);
+        for(int fieldNum=5; fieldNum<=NumAlphas; fieldNum++) {
+            if(cAlphaArgs(fieldNum) == "") {
+                break;
+            }
+            input_specs.speed_data_names.push_back(cAlphaArgs(fieldNum));
+        }
+
+        this->instantiateFromInputSpec(input_specs);
+    }
+
+    if (!found_it) {
+        // error
+    }
+
+}
+
+void CoilCoolingDXCurveFitOperatingMode::sizeOperatingMode() {
+
+    std::string RoutineName = "sizeOperatingMode";
+    std::string CompType = this->object_name;
+    std::string CompName = this->name;
+    bool PrintFlag = true;
+
+    // Set speed objects parent pointer (this is kind of a test - probably needs to be a separate init function)
+    for (auto & curSpeed : this->speeds) {
+        curSpeed.parentMode = this;
+    }
```

Comment:
@Myoldmopar Moved assignment of parentMode to here so that it's done after the operating mode object has been fully built and emplaced. Seems to solve the moving pointer problem and gets past the sizing error. Not sure if this is the best spot to put this.